Build failure on CPython 3.13 (dev)

src/gmpy2_hash.c:219:27: error: ‘_PyHASH_IMAG’ undeclared:
```
$ pip install .
Processing /home/sk/src/gmpy
  Installing build dependencies ... done
  Getting requirements to build wheel ... done
  Preparing metadata (pyproject.toml) ... done
Building wheels for collected packages: gmpy2
  Building wheel for gmpy2 (pyproject.toml) ... error
  error: subprocess-exited-with-error

  × Building wheel for gmpy2 (pyproject.toml) did not run successfully.
  │ exit code: 1
  ╰─> [43 lines of output]
      running bdist_wheel
      running build
      running build_py
      running egg_info
      writing gmpy2.egg-info/PKG-INFO
      writing dependency_links to gmpy2.egg-info/dependency_links.txt
      writing requirements to gmpy2.egg-info/requires.txt
      writing top-level names to gmpy2.egg-info/top_level.txt
      reading manifest file 'gmpy2.egg-info/SOURCES.txt'
      reading manifest template 'MANIFEST.in'
      adding license file 'COPYING'
      adding license file 'COPYING.LESSER'
      writing manifest file 'gmpy2.egg-info/SOURCES.txt'
      running build_ext
      building 'gmpy2.gmpy2' extension
      gcc -fno-strict-overflow -Wsign-compare -DNDEBUG -g -O3 -Wall -fPIC -I./src -I/home/sk/src/cpython/for-gmpy/include -I/home/sk/src/cpython/Include -I/home/sk/src/cpython -c src/gmpy2.c -o build/temp.linux-x86_64-cpython-313/src/gmpy2.o -DSHARED=1
      In file included from src/gmpy2.c:621:
      src/gmpy2_convert_gmp.c: In function ‘GMPy_PyLong_From_MPZ’:
      src/gmpy2_convert_gmp.c:160:20: warning: implicit declaration of function ‘_PyLong_New’ [-Wimplicit-function-declaration]
        160 |     if (!(result = _PyLong_New(size))) {
            |                    ^~~~~~~~~~~
      src/gmpy2_convert_gmp.c:160:18: warning: assignment to ‘PyLongObject *’ {aka ‘struct _longobject *’} from ‘int’ makes pointer from integer without a cast [-Wint-conversion]
        160 |     if (!(result = _PyLong_New(size))) {
            |                  ^
      In file included from src/gmpy2.c:623:
      src/gmpy2_convert_mpc.c: In function ‘GMPy_MPC_From_PyStr’:
      src/gmpy2_convert_mpc.c:281:13: warning: assignment discards ‘const’ qualifier from pointer target type [-Wdiscarded-qualifiers]
        281 |         cp  = PyUnicode_AsUTF8AndSize(s, &len);
            |             ^
      In file included from src/gmpy2.c:664:
      src/gmpy2_hash.c: In function ‘_mpfr_hash’:
      src/gmpy2_hash.c:187:5: warning: returning ‘void *’ from a function with return type ‘Py_hash_t’ {aka ‘long int’} makes integer from pointer without a cast [-Wint-conversion]
        187 |     CHECK_CONTEXT(context);
            |     ^~~~~~~~~~~~~ 
      src/gmpy2_hash.c:189:12: warning: implicit declaration of function ‘_Py_HashDouble’ [-Wimplicit-function-declaration]
        189 |     return _Py_HashDouble(temp);
            |            ^~~~~~~~~~~~~~
      src/gmpy2_hash.c: In function ‘GMPy_MPC_Hash_Slot’:
      src/gmpy2_hash.c:219:27: error: ‘_PyHASH_IMAG’ undeclared (first use in this function)
        219 |     combined = hashreal + _PyHASH_IMAG * hashimag;
            |                           ^~~~~~~~~~~~
      src/gmpy2_hash.c:219:27: note: each undeclared identifier is reported only once for each function it appears in
      error: command '/usr/bin/gcc' failed with exit code 1
      [end of output]

  note: This error originates from a subprocess, and is likely not a problem with pip.
  ERROR: Failed building wheel for gmpy2
Failed to build gmpy2
ERROR: Could not build wheels for gmpy2, which is required to install pyproject.toml-based projects
```

--- a/.github/workflows/pip_install_gmpy2.yml
+++ b/.github/workflows/pip_install_gmpy2.yml
@@ -7,7 +7,7 @@ jobs:
     strategy:
       fail-fast: false
       matrix:
-        python-version: [3.11, 3.12]
+        python-version: [3.11, 3.12, 3.13]
         os: [macos-latest]
     runs-on: ${{ matrix.os }}
     steps:

--- a/src/gmpy2_convert.h
+++ b/src/gmpy2_convert.h
@@ -154,6 +154,44 @@ extern "C" {
 #  define _PyLong_DigitCount(obj) (_PyLong_IsNegative(obj)? -Py_SIZE(obj):Py_SIZE(obj))
 #endif
 
+#if PY_VERSION_HEX >= 0x030D0000
+
+#define MAX_LONG_DIGITS \
+    ((PY_SSIZE_T_MAX - offsetof(PyLongObject, long_value.ob_digit))/sizeof(digit))
+
+PyLongObject *
+_PyLong_New(Py_ssize_t size)
+{
+    assert(size >= 0);
+    PyLongObject *result;
+    if (size > (Py_ssize_t)MAX_LONG_DIGITS) {
+        PyErr_SetString(PyExc_OverflowError,
+                        "too many digits in integer");
+        return NULL;
+    }
+    /* Fast operations for single digit integers (including zero)
+     * assume that there is always at least one digit present. */
+    Py_ssize_t ndigits = size ? size : 1;
+    /* Number of bytes needed is: offsetof(PyLongObject, ob_digit) +
+       sizeof(digit)*size.  Previous incarnations of this code used
+       sizeof() instead of the offsetof, but this risks being
+       incorrect in the presence of padding between the header
+       and the digits. */
+    result = PyObject_Malloc(offsetof(PyLongObject, long_value.ob_digit) +
+                             ndigits*sizeof(digit));
+    if (!result) {
+        PyErr_NoMemory();
+        return NULL;
+    }
+    _PyLong_SetSignAndDigitCount(result, size != 0, size);
+    PyObject_Init((PyObject*)result, &PyLong_Type);
+    /* The digit has to be initialized explicitly to avoid
+     * use-of-uninitialized-value. */
+    result->long_value.ob_digit[0] = 0;
+    return result;
+}
+#endif
+
 /* Since the macros are used in gmpy2's codebase, these functions are skipped
  * until they are needed for the C API in the future.
  */

--- a/src/gmpy2_hash.c
+++ b/src/gmpy2_hash.c
@@ -24,6 +24,11 @@
  * License along with GMPY2; if not, see <http://www.gnu.org/licenses/>    *
  * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * * */
 
+#if PY_VERSION_HEX >= 0x030D0000
+# define Py_BUILD_CORE
+# include <internal/pycore_pyhash.h>
+#endif
+
 static Py_hash_t
 GMPy_MPZ_Hash_Slot(MPZ_Object *self)
 {

